### PR TITLE
remove ObjectMeta and *Options from core api

### LINKED ISSUE #61807: It's time to remove ObjectMeta and *Options in core api
https://github.com/kubernetes/kubernetes/pull/39709 has moved ObjectMeta  to meta/v1 and mark it deprecated in core api.
https://github.com/kubernetes/kubernetes/pull/40344 has moved ListOptions DeleteOptions to meta/v1 and mark it deprecated in core api.

So should remove it from core api group.

/sig api-machinery

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
remove ObjectMeta and *Options from core api

Fixes #61807

**Special notes for your reviewer**:

**Release note**:

```release-note
Remove `ObjectMeta ` `ListOptions` `DeleteOptions` from core api group.  Please use that in meta/v1
```

### CODE REVIEW

File: pkg/api/testing/defaulting_test.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ func TestDefaulting(t *testing.T) {
 			c.FuzzNoCustom(s)
 			s.MatchExpressions = nil // need to fuzz this specially
 		},
-		func(s *apiv1.ListOptions, c fuzz.Continue) {
+		func(s *metav1.ListOptions, c fuzz.Continue) {
```
Comment (deads2k) [inline pkg/api/testing/defaulting_test.go]: check for an existing fuzzer in apimachinery for this

**Comment by deads2k** [comment]
comment on test, squash the generated changes into a single generation commit.

@smarterclayton for approval

**Comment by hzxuzhonghu** [comment]
squashed

**Comment by hzxuzhonghu** [comment]
ping @smarterclayton for approval

**Comment by hzxuzhonghu** [comment]
@deads2k need a lgtm tag

**Comment by deads2k** [comment]
/lgtm
/hold

holding for squash, feel free to remove after squashing.

**Pull request merged**